test/index.js

```diff
@@ -0,0 +1,141 @@
+/* global describe, it */
+import assert from 'assert';
+import SimpleTestServer from '../';
+import http from 'http';
+
+describe('SimpleTestServer', function () {
+	let server;
+	it('should should load the SimpleTestServer class', function () {
+		assert.ok(SimpleTestServer, 'SimpleTestServer class loaded');
+		server = new SimpleTestServer();
+		assert.ok(server, 'server initiated');
+	});
+	it('should should start the server', function (done) {
+		server.start(done);
+	});
+	it('should register the port', function () {
+		assert(typeof server.port === 'number', 'server.port is a number');
+	});
+	it('should make a GET request', function (done) {
+		const httpOpts = {
+			method: 'GET',
+			port: server.port,
+			path: '/a/path?q=findme'
+		};
+		const req = http.request(httpOpts, function (res) {
+			let body = '';
+			res.on('data', function (chunk) {
```

Comment:
Request can do all of this for you, it will parse the json, accumulate the body and properly get any errors.